csrc/python_frontend/fusion_record.h

```diff
@@ -1229,15 +1229,24 @@ struct TensorRecord : RecordFunctor {
     auto rank = shape_.size();
     std::vector<bool> is_expand(rank);
 
-    for (const auto contig_index : c10::irange(rank)) {
-      bool is_broadcast = !contiguity_[contig_index].has_value();
+    for (const auto index : c10::irange(rank)) {
       // since contiguity_ vector is given to the corresponding order in alloc
       // domain, while is_expand is given to root domain, we need to map it
       // correctly with `contig_index` and `index`.
-      const auto index = stride_order_.empty()
-          ? contig_index
-          : rank - 1 - static_cast<size_t>(stride_order_[contig_index]);
+      //
+      // stride_order[i] indicates that:
+      //   `root_domain[i]` maps to `alloc_domain[rank - 1 - stride_order_[i]]`
+      //
+      // Hence `index` on root domain would be corresponding to the contiguity
+      // index `contig_index = rank - 1 - stride_order[index]`
+      const auto contig_index = stride_order_.empty()
+          ? index
+          : rank - 1 - static_cast<size_t>(stride_order_[index]);
+      bool is_broadcast = !contiguity_[contig_index].has_value();
       bool has_symbolic_size = (shape_[index] == -1);
+      // A root dimension is expand dimension if:
+      //   The dimension is marked a broadcast; and
+      //   The dimension has a symbolic size (expand extend)
```

Comment:
Wait, is this right? I'm pretty sure I've seen a broadcast dimension with a constant expand extent.

Reply:
hmmm. That's a good catch... I guess the real case that I need to check is `not broadcast` i.e. `!= -1`?!

Let me try some more tests.

Reply:
You caught me there.

Fixed the logic here and added another entry in the python test to verify that.

Reply:
Did you forget to `git push` your latest changes?

Reply:
Oops messed up my history. just pushed.